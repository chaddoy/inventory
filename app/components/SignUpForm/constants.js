export const SUCCESS_MSG = 'Registration Successful!';
export const SUCCESS_DESC = `
  Please check your email in a few minutes to verify your account.
  If you have not recieved it, please check your spam folder, then mark it as
  NOT spam.
`;
export const FORM_ITEM_LAYOUT = {
  labelCol: {
    xs: {
      span: 1,
      offset: 0,
    },
    sm: {
      span: 6,
      offset: 0,
    },
    md: {
      span: 5,
      offset: 0,
    },
    lg: {
      span: 5,
      offset: 0,
    },
  },
  wrapperCol: {
    xs: {
      span: 24,
      offset: 0,
    },
    sm: {
      span: 17,
      offset: 1,
    },
    md: {
      span: 19,
      offset: 0,
    },
    lg: {
      span: 19,
      offset: 0,
    },
  },
};
