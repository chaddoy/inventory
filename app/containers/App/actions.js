/*
 *
 * LoginPage actions
 *
 */

import {
  CHECK_USER_AUTH,
  SET_USER_AUTH,
  ERROR_USER_AUTH,
} from './constants';

export function checkUserAuth({ username, password }) {
  return {
    type: CHECK_USER_AUTH,
    username,
    password,
  };
}

export function setUserAuth(user) {
  return {
    type: SET_USER_AUTH,
    user,
  };
}

export function errUserAuth(errorMsg) {
  return {
    type: ERROR_USER_AUTH,
    errorMsg,
  };
}
