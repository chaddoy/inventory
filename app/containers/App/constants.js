/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const DEFAULT_LOCALE = 'en';
export const CHECK_USER_AUTH = 'app/App/CHECK_USER_AUTH';
export const SET_USER_AUTH = 'app/App/SET_USER_AUTH';
export const ERROR_USER_AUTH = 'app/App/ERROR_USER_AUTH';
export const UNSET_USER_AUTH = 'app/App/UNSET_USER_AUTH';

export const NAVS = [{
  path: '/',
  displayName: 'Dashboard',
  name: 'dashboard',
  icon: 'dashboard',
}, {
  path: '/users',
  displayName: 'Users',
  name: 'users',
  icon: 'user-o',
}];
