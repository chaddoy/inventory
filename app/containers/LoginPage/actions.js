/*
 *
 * LoginPage actions
 *
 */

import {
  LOGIN,
  LOGIN_USER,
  LOGIN_ERROR,
} from './constants';

export function login({ username, password }) {
  return {
    type: LOGIN,
    username,
    password,
  };
}

export function loginUser(user) {
  return {
    type: LOGIN_USER,
    user,
  };
}

export function loginError(errorMsg) {
  return {
    type: LOGIN_ERROR,
    errorMsg,
  };
}
