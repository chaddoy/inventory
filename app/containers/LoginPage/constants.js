/*
 *
 * LoginPage constants
 *
 */

export const LOGIN = 'app/LoginPage/LOGIN';
export const LOGIN_USER = 'app/LoginPage/LOGIN_USER';
export const LOGIN_ERROR = 'app/LoginPage/LOGIN_ERROR';

export const USERS = [{
  username: 'admin',
  password: 'admin',
  firstName: 'Abraham',
  lastName: 'Maslow',
}, {
  username: 'r.botero',
  password: 'secret',
  firstName: 'Rommel',
  lastName: 'Botero',
}];

// MESSAGES
export const LOGIN_ERROR_MSG = 'Wrong username or password';
