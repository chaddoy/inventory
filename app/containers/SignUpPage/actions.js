/*
 *
 * SignUpPage actions
 *
 */

import {
  SIGNUP_USER,
  SIGNUP_USER_SUCCESS,
  SIGNUP_USER_ERROR,
} from './constants';

export function signUpUser({ email, password }) {
  return {
    type: SIGNUP_USER,
    email,
    password,
  };
}

export function signUpUserSuccess() {
  return {
    type: SIGNUP_USER_SUCCESS,
  };
}

export function signUpUserError(errorMsg) {
  return {
    type: SIGNUP_USER_ERROR,
    errorMsg,
  };
}
