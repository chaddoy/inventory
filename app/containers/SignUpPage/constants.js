/*
 *
 * SignUpPage constants
 *
 */

export const SIGNUP_USER = 'app/SignUpPage/SIGNUP_USER';
export const SIGNUP_USER_SUCCESS = 'app/SignUpPage/SIGNUP_USER_SUCCESS';
export const SIGNUP_USER_ERROR = 'app/SignUpPage/SIGNUP_USER_ERROR';

export const SIGNUP_SUCCESS_MSG = `
  Thank you for registering.
  Please verify your account.
`;
