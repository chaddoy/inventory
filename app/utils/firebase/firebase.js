import * as firebase from 'firebase';

const prodConfig = {
  apiKey: 'AIzaSyC14TDzm3SSt8VqBQYx-DEc7t-4p7Z4rRo',
  authDomain: 'inventory-service-70274.firebaseapp.com',
  databaseURL: 'https://inventory-service-70274.firebaseio.com',
  projectId: 'inventory-service-70274',
  storageBucket: 'inventory-service-70274.appspot.com',
  messagingSenderId: '189137843330',
};

const devConfig = {
  apiKey: 'AIzaSyDOuyT0zESVhBxRcVvdGEtMq2YVcys3rQI',
  authDomain: 'inventory-service-b1875.firebaseapp.com',
  databaseURL: 'https://inventory-service-b1875.firebaseio.com',
  projectId: 'inventory-service-b1875',
  storageBucket: 'inventory-service-b1875.appspot.com',
  messagingSenderId: '913357789158',
};

const config = process.env.NODE_ENV === 'production'
  ? prodConfig
  : devConfig;

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

export {
  auth,
};
