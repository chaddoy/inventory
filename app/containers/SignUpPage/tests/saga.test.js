/**
 * Tests for HomePage sagas
 */

import { call, put, takeEvery } from 'redux-saga/effects';

import { auth } from 'utils/firebase';
import { signUpUserSuccess, signUpUserError } from '../actions';
import { SIGNUP_USER } from '../constants';
import watchSignUp, {
  registerUser,
  addUserToFireDB,
} from '../saga';

const user = {
  username: 'janedoe',
  firstName: 'Jane',
  middleName: 'Foo',
  lastName: 'Doe',
  address: 'matrix',
  gender: 'male',
  phoneNumber: '1234567890',
  email: 'janedoe@email.com',
  password: 'myPassword',
  confirmPassword: 'myPassword',
};

const johndoe = Object.assign({}, user, {
  username: 'johndoe',
  firstName: 'John',
  email: 'johndoe@email.com',
});

const users = {
  val: () => ({
    johndoe,
  }),
};

/* eslint-disable redux-saga/yield-effects */
describe('registerUser Saga', () => {
  let registerUserGenerator;

  beforeEach(() => {
    registerUserGenerator = registerUser({ user });

    const getUsersDescriptor = registerUserGenerator.next().value;
    expect(getUsersDescriptor).toMatchSnapshot();
  });

  it('should dispatch the signUpUserSuccess action if it requests the data successfully', () => {
    const callDBDescriptor = registerUserGenerator.next(users).value;
    expect(callDBDescriptor).toEqual(call(addUserToFireDB, user));

    const callAuthDescriptor = registerUserGenerator.next(users).value;
    expect(callAuthDescriptor).toEqual(call(
      auth.doCreateUserWithEmailAndPassword,
      user.email,
      user.password
    ));

    const putDescriptor = registerUserGenerator.next(user).value;
    expect(putDescriptor).toEqual(put(signUpUserSuccess('')));
  });

  it('should throw a message if email exists', () => {
    const putDescriptor = registerUserGenerator.next({
      val: () => ({
        janedoe: user,
      }),
    }).value;
    expect(putDescriptor).toEqual(put(signUpUserError('Email already exist')));
  });

  it('should throw a message if username exists', () => {
    const putDescriptor = registerUserGenerator.next({
      val: () => ({
        janedoe: Object.assign({}, user, {
          email: 'new@email.com',
        }),
      }),
    }).value;
    expect(putDescriptor).toEqual(put(signUpUserError('Username already exist')));
  });

  it('should call the signUpUserError action if the response errors', () => {
    const response = { message: 'Something wnet wrong.' };
    const putDescriptor = registerUserGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(signUpUserError(response.message)));
  });
});

describe('watchSignUpSaga Saga', () => {
  const watchSignUpSaga = watchSignUp();

  it('should start task to watch for SIGNUP_USER action', () => {
    const takeEveryDescriptor = watchSignUpSaga.next(user).value;
    expect(takeEveryDescriptor).toEqual(takeEvery(SIGNUP_USER, registerUser));
  });
});
