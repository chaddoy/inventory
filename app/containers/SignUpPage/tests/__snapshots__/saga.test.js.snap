// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerUser Saga should call the signUpUserError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`registerUser Saga should dispatch the signUpUserSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`registerUser Saga should throw a message if email exists 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`registerUser Saga should throw a message if username exists 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
